gnovm/stdlibs/testing/fuzz.gno

```diff
@@ -1,284 +1,546 @@
+	// This completes the status change of the seed before logging
+	hashNumber := f.fhm.RegisterCoverage2HashNumber(coverage)
+	seed.HashNumber = hashNumber
+	if isPanic {
+		tr := TestResult{
+			PanicOccurred: true,
+			PanicMessage:  panicMsg,
+			Error:         err,
 		}
-
-		run(nil, args...)
+		f.fsm.Crash_Logger.AddCase(seed, tr)
+		f.Fail()
 	}
-}
-
-// Add adds test values to initialize the corpus.
-func (f *F) Add(values ...interface{}) []Fuzzer {
-	fuzzers := make([]Fuzzer, len(values))
-
-	for i, v := range values {
-		str, ok := v.(string)
-		if !ok {
-			continue
+	if err != nil {
+		tr := TestResult{
+			PanicOccurred: false,
+			PanicMessage:  "",
+			Error:         err,
 		}
-		f.corpus = append(f.corpus, str)
-		fuzzers[i] = &StringFuzzer{Value: str}
+		f.fsm.Crash_Logger.AddCase(seed, tr)
+		f.Fail()
 	}
+	return seed
+}
 
-	return fuzzers
+// Fail marks the function as having failed bur continue execution.
+func (f *F) Fail() {
+	f.failed = true
 }
 
-func (f *F) evolve(generations int) {
-	population := make([]*Individual, len(f.corpus))
-	for i, c := range f.corpus {
-		population[i] = &Individual{Fuzzer: &StringFuzzer{Value: c, f: f}}
-	}
+// Fatal is equivalent to Log followed by FailNow.
+// It logs the message and marks the fuzzing as failed.
+func (f *F) Fatal(args ...interface{}) {
+	var sb strings.Builder
 
-	for _, ind := range population {
-		ind.calculateFitness()
+	for _, arg := range args {
+		sb.WriteString(arg.(string))
 	}
 
-	for gen := 0; gen < generations; gen++ {
-		population = Selection(population)
-		newPopulation := make([]*Individual, 0, len(population))
+	f.msgs = append(f.msgs, sb.String())
+	f.Fail()
+}
+
+func (f *F) checkProgress(content []interface{}, minimazalbeIdXs []int) ([]interface{}, bool, int) {
+	changed := false
+	for _, idx := range minimazalbeIdXs {
+		sOrb := content[idx]
 
-		for i := 0; i < len(population); i += 2 {
-			if i+1 < len(population) {
-				child1, child2 := Crossover(population[i], population[i+1])
-				newPopulation = append(newPopulation, child1, child2)
+		switch v := sOrb.(type) {
+		case string:
+			if len(v) < 1 {
 				continue
 			}
-
-			newPopulation = append(newPopulation, population[i])
+			for i := 0; i < len(v); i++ {
+				b := []byte(v)
+				candidate := append(b[:i], b[i+1:]...)
+
+				tester := make([]interface{}, len(content))
+				for i, v := range content {
+					tester[i] = v
+				}
+				tester[idx] = string(candidate)
+				if f.checkPreserve(tester).errorIsPreserved {
+
+					changed = true
+					return tester, changed, idx
+				}
+			}
+		case []byte:
+			if len(v) < 1 {
+				continue
+			}
+			for i := 0; i < len(v); i++ {
+				b := []byte(v)
+				candidate := append(b[:i], b[i+1:]...)
+				tester := make([]interface{}, len(content))
+				for i, v := range content {
+					tester[i] = v
+				}
+				tester[idx] = []byte(candidate)
+				if f.checkPreserve(tester).errorIsPreserved {
+					changed = true
+					return tester, changed, idx
+				}
+			}
```

Comment:
These lines are almost same for `string` and `[]byte` type. I think you can extracting this as a function.